VideoLocker/src/main/java/org/edx/mobile/task/SetThreadReadTask.java

```diff
@@ -11,22 +11,15 @@
 
 public class SetThreadReadTask extends Task<DiscussionThread> {
     private final DiscussionThread thread;
-    private final boolean read;
 
     public SetThreadReadTask(@NonNull Context context,
-                             @NonNull DiscussionThread thread, boolean read) {
+                             @NonNull DiscussionThread thread) {
         super(context);
         this.thread = thread;
-        this.read = read;
     }
 
     @Override
     public DiscussionThread call() throws RetroHttpException {
-        return environment.getDiscussionAPI().setThreadRead(thread, read);
-    }
-
-    @Override
-    protected void onSuccess(DiscussionThread discussionThread) {
-        EventBus.getDefault().post(new DiscussionThreadUpdatedEvent(discussionThread));
+        return environment.getDiscussionAPI().setThreadRead(thread, true);
```

Comment:
OK, how about renaming it to `GetAndReadThreadTask` to make it both unambiguous, and still convey the main purpose that it's being used for?